sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/BeamFnDataGrpcClient.java

```diff
@@ -80,7 +79,7 @@ public BeamFnDataGrpcClient(
       Endpoints.ApiServiceDescriptor apiServiceDescriptor,
```

Comment:
I never thought the BeamFnDataInboundObserver would need to be responsible for closing the consumers as the consumers were expected to be stateless. The only one that needed some state was the outbound consumer since it needs a signal of when the bundle processing is done. This is why the ThrowingConsumer / CloseableThrowingConsumer differentiation existed on forInbound/forOutbound. In reality the closing part is meant to be handled by the ProcessBundleHandler and the consuming/throwing part is meant to be handled by an upstream "receiver".

They are really supposed to model two different objects but multiple returns in Java are annoying. Maybe we could return <? extends AutoCloseable & FnDataReceiver> and get rid of the close method on FnDataReceiver or alternatively the inboundObserver should take something else then a FnDataReceiver.

Reply:
Re-forked the two interfaces.